### PR TITLE
!!! [v6r12] List transformations by UserName/AuthorDN

### PR DESCRIPTION
**Pull request by calancha**
Added functions:
[1] do_getallByUser
(DIRAC/TransformationSystem/Client/TransformationCLI.py)

[2] getAuthorDNfromProxy
[3] getTransformationsByUser
(DIRAC/TransformationSystem/Client/Transformation.py)

Monitoring ilcdirac productions from the terminal presents a difficulty:
users do not own their productions: all productions own to the leader
of this project. This cause that users can not list their productions using
their username as a filter

To overcome this problem a dedicated function has being introduced in the
transformation CLI: getallByUser

This function allow list productions filtering by AuthorDN. Then, getTransformationsByUser
is called.

Because AuthorDN is hard to remember, getallByUser also
accept the dirac username. When username is provided instead of the AuthorDN,
another function (getAuthorDNfromProxy) gets the AuthorDN from the current uploaded
proxy: only if the username of the proxy matchs the one provided in the command line,
this AuthorDN is used.

Examples:

```
(Cmd) getallByUser '/DC=ch/DC=cern/OU=Organic Units/OU=Users/CN=sposs/CN=641989/CN=Stephane Guillaume Poss' Stopped
TransformationID      Status       AgentType      TransformationName                                 CreationDate             AuthorDN                                                                                   
2340                  Stopped      Manual         ee_nnll_1400.0_cut_cut_beamrecoil                  2013-04-17 13:55:24      /DC=ch/DC=cern/OU=Organic Units/OU=Users/CN=sposs/CN=641989/CN=Stephane Guillaume Poss     
2341                  Stopped      Manual         ee_nnll_1400.0_sid_sim_cut_beamrecoil              2013-04-17 13:55:27      /DC=ch/DC=cern/OU=Organic Units/OU=Users/CN=sposs/CN=641989/CN=Stephane Guillaume Poss     
2342                  Stopped      Manual         ee_nnll_1400.0_sid_rec_overlay_cut_beamrecoil      2013-04-17 13:55:31      /DC=ch/DC=cern/OU=Organic Units/OU=Users/CN=sposs/CN=641989/CN=Stephane Guillaume Poss     
(Cmd) 
```

```
(Cmd) getallByUser calanchac Stopped
TransformationID      Status       AgentType      TransformationName                                        CreationDate             AuthorDN                                       
5893                  Stopped      Manual         37588_1000.0_split_test-1000gev_20151007_1                2015-10-07 12:46:54      /C=JP/O=KEK/OU=CRC/CN=CALANCHA Constantino     
5894                  Stopped      Manual         37588_1000.0_ild_sim_test-1000gev_20151007_1              2015-10-07 12:47:44      /C=JP/O=KEK/OU=CRC/CN=CALANCHA Constantino     
5895                  Stopped      Manual         37588_1000.0_ild_rec_overlay_test-1000gev_20151007_1      2015-10-07 12:48:32      /C=JP/O=KEK/OU=CRC/CN=CALANCHA Constantino     
5899                  Stopped      Manual         108088_350.0_split_test-1000gev_20151007_1                2015-10-07 13:09:30      /C=JP/O=KEK/OU=CRC/CN=CALANCHA Constantino     
5960                  Stopped      Manual         108168_500.0_split_test-1000gev_20151008_1                2015-10-08 02:23:07      /C=JP/O=KEK/OU=CRC/CN=CALANCHA Constantino     
5993                  Stopped      Manual         108709_350.0_ild_sim_20151018_1                           2015-10-18 11:40:19      /C=JP/O=KEK/OU=CRC/CN=CALANCHA Constantino     
5997                  Stopped      Manual         108710_350.0_ild_sim_20151018_1                           2015-10-18 11:40:55      /C=JP/O=KEK/OU=CRC/CN=CALANCHA Constantino     
6000                  Stopped      Manual         108709_350.0_ild_rec_overlay_20151018_1                   2015-10-18 11:41:10      /C=JP/O=KEK/OU=CRC/CN=CALANCHA Constantino     
6001                  Stopped      Manual         108712_350.0_ild_sim_20151018_1                           2015-10-18 11:41:17      /C=JP/O=KEK/OU=CRC/CN=CALANCHA Constantino     
6003                  Stopped      Manual         108707_350.0_ild_sim_20151018_1                           2015-10-18 11:41:28      /C=JP/O=KEK/OU=CRC/CN=CALANCHA Constantino     
6005                  Stopped      Manual         108708_350.0_ild_sim_20151018_1                           2015-10-18 11:41:35      /C=JP/O=KEK/OU=CRC/CN=CALANCHA Constantino     
6008                  Stopped      Manual         108701_500.0_ild_sim_20151018_1                           2015-10-18 11:41:52      /C=JP/O=KEK/OU=CRC/CN=CALANCHA Constantino     
6011                  Stopped      Manual         108706_500.0_ild_sim_20151018_1                           2015-10-18 11:42:08      /C=JP/O=KEK/OU=CRC/CN=CALANCHA Constantino     
6014                  Stopped      Manual         108705_500.0_ild_sim_20151018_1                           2015-10-18 11:42:24      /C=JP/O=KEK/OU=CRC/CN=CALANCHA Constantino     
6015                  Stopped      Manual         108703_500.0_ild_sim_20151018_1                           2015-10-18 11:42:32      /C=JP/O=KEK/OU=CRC/CN=CALANCHA Constantino     
6016                  Stopped      Manual         108701_500.0_ild_rec_overlay_20151018_1                   2015-10-18 11:42:40      /C=JP/O=KEK/OU=CRC/CN=CALANCHA Constantino     
6017                  Stopped      Manual         108706_500.0_ild_rec_overlay_20151018_1                   2015-10-18 11:42:56      /C=JP/O=KEK/OU=CRC/CN=CALANCHA Constantino     
6018                  Stopped      Manual         108705_500.0_ild_rec_overlay_20151018_1                   2015-10-18 11:43:12      /C=JP/O=KEK/OU=CRC/CN=CALANCHA Constantino     
6019                  Stopped      Manual         108703_500.0_ild_rec_overlay_20151018_1                   2015-10-18 11:43:20      /C=JP/O=KEK/OU=CRC/CN=CALANCHA Constantino     
6026                  Stopped      Manual         108704_500.0_split_20151020_1                             2015-10-20 06:48:12      /C=JP/O=KEK/OU=CRC/CN=CALANCHA Constantino     
6027                  Stopped      Manual         108704_500.0_ild_sim_20151020_1                           2015-10-20 06:49:01      /C=JP/O=KEK/OU=CRC/CN=CALANCHA Constantino     
6028                  Stopped      Manual         108704_500.0_ild_rec_overlay_20151020_1                   2015-10-20 06:49:50      /C=JP/O=KEK/OU=CRC/CN=CALANCHA Constantino     
(Cmd) 
```

```
(Cmd) getallByUser sposs Stopped
Couldn't resolve the authorDN for user 'sposs' from the uploaded proxy (proxy created by 'calanchac') 
(Cmd) 
```

### CODE REVIEW

**Comment by andresailer** [comment]
Please move the offtopic parts to the ilcdirac jira.

You need to create your new feature branch properly. At the moment this contains commits that do not belong.

File: TransformationSystem/Client/TransformationCLI.py [hunk 1]
```diff
@@ -14,6 +14,7 @@
 
 from DIRAC.TransformationSystem.Client.Transformation           import Transformation
 from DIRAC.TransformationSystem.Client.TransformationClient     import TransformationClient
+from DIRAC.Resources.Catalog.FileCatalogClient                  import FileCatalogClient
```
Comment (andresailer) [inline TransformationSystem/Client/TransformationCLI.py]: You are not using FileCatalogClient, are you?

Comment (calancha) [inline TransformationSystem/Client/TransformationCLI.py]: You are right. It seems i used in a previous version that i superseded.
I am leaving now office. I look your comments tomorrow morning.

Comment (fstagni) [inline TransformationSystem/Client/TransformationCLI.py]: I would suggest running pylint on these files, it'd spot these type of things immediately.

**Comment by calancha** [comment]
Show Show getallByUser (instead of do_getallByUser) in the 
documentation string of the function.

5997f16f3611fdea3888f22349bfc19f0e498ef

File: TransformationSystem/Client/TransformationCLI.py (hunk 1 above)
Comment (atsareg) [inline TransformationSystem/Client/TransformationCLI.py]: You have to redo this PR please. By procedure I can not pick up new changes from a commit in some you branch. So, create a branch which contains the necessary changes with respect to the main rel-v6r12 branch. Push this branch to your forked repository in GitHub. Make a new PR against the rel-v6r12 branch.

**Comment by calancha** [comment]
Closing this PR because cannot be merged.
Supersed by PR: https://github.com/DIRACGrid/DIRAC/pull/2687

**Pull request closed without merging**